src/libsync/propagateuploadencrypted.h

```diff
@@ -70,6 +71,9 @@ private slots:
   FolderMetadata *_metadata;
   EncryptedFile _encryptedFile;
   QString _completeFileName;
+
+  bool _isUnlockRunning = false; // protection against multiple calls to unlock the same folder
+  QMutex _isUnlockRunningMutex; // corresponding mutex for _isUnlockRunning
```

Comment:
I'm unsure why we have a mutex here, it's all in the main thread anyway, isn't it?

Reply:
@er-vin I am assuming each job has its own thread? But, I will double-check that.

Reply:
No they don't they all run on the main thread event loop. This is asynchronous thanks to the event loop but not multi-threaded.